### ISSUE TITLE
User categories are not supported for non installed games.

### ISSUE THREAD

**Issue by BeatLink** [lutris/lutris#5123]
### Bug description

Games that are in the Lutris library but not yet installed cannot be added to a specific Category.

### How to Reproduce

1. Find a game in the Lutris library that is not yet installed
2. Right click on the game
3. Note that there is no Categories menu that would otherwise be shown on installed games

### Expected behavior

There should be a categories menu item that lets you set the category for the game

### Log output

```shell
2023-11-09 08:33:49,947: Command 'fluidsynth' not found on your system
INFO     2023-11-09 08:33:50,328 [application.do_command_line:436]:Starting Lutris 0.5.14
INFO     2023-11-09 08:33:50,330 [startup.get_drivers:57]:Using NVIDIA drivers 535.129.03 for x86_64
INFO     2023-11-09 08:33:50,330 [startup.get_drivers:61]:GPU: NVIDIA GeForce RTX 3050 Ti Laptop GPU
INFO     2023-11-09 08:33:50,330 [display.display_gpu_info:78]:GPU: 10DE:25A0 17AA:3A5D (nvidia drivers)
INFO     2023-11-09 08:33:50,331 [display.display_gpu_info:78]:GPU: 1002:1638 17AA:3A5D (amdgpu drivers)
DEBUG    2023-11-09 08:33:50,770 [xrandr._get_vidmodes:16]:Retrieving video modes from XrandR
DEBUG    2023-11-09 08:33:51,504 [lutris.get_missing_game_ids:202]:Checking for missing games
DEBUG    2023-11-09 08:34:05,482 [services.get_game:34]:Getting service game 300 for steam
DEBUG    2023-11-09 08:34:05,484 [services.get_game:34]:Getting service game 300 for steam
```

### System Information

```shell
[System]
OS:              GNOME 45 Flatpak runtime
Arch:            x86_64
Kernel:          6.2.0-36-generic
Desktop:         X-Cinnamon
Display Server:  x11

[CPU]
Vendor:          AuthenticAMD
Model:           AMD Ryzen 5 5600H with Radeon Graphics
Physical cores:  6
Logical cores:   12

[Memory]
RAM:             13.5 GB
Swap:            1.0 GB

[Graphics]
Vendor:          AMD
OpenGL Renderer: AMD Radeon Graphics (renoir, LLVM 16.0.6, DRM 3.49, 6.2.0-36-generic)
OpenGL Version:  4.6 (Compatibility Profile) Mesa 23.2.1 (git-49a47f187e)
OpenGL Core:     4.6 (Core Profile) Mesa 23.2.1 (git-49a47f187e)
OpenGL ES:       OpenGL ES 3.2 Mesa 23.2.1 (git-49a47f187e)
Vulkan Version:  1.3.261
Vulkan Drivers:  AMD Radeon Graphics (RADV RENOIR) (1.3.255), NVIDIA GeForce RTX 3050 Ti Laptop GPU (1.3.242)
```

### Checklist:

- [X] I'm not asking for support with a game or the wine runner.
- [X] I have followed the above mentioned guides and have all the graphics and wine dependencies installed.
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

**Comment by danieljohnson2** [comment]
Looking into this, it's kinda funny. You *can* have categories on games that are not installed, so long as they are in your PGA.

Here's how to do it. Install the game, start it, give it a category, then remove it. It will become uninstalled, but the icon is still there (but faded out). It does not have a  "Categories" command, but the category you set up still exists and still lists the game.

I think that is a pretty useless thing, but it does illustrate the issue- you can't have  categories until the game is in the local database, and if it is, it appears in your "Games" view. I suspect you would not desire this, even if it were less awkward to achieve.

**Comment by strycore** [comment]
@danieljohnson2 looks like you're over-complicating things a bit, or mixing service games with games ^^

**Comment by danieljohnson2** [comment]
Maybe. But I just tried your patch, and I can't actually put an entirely uninstalled game into a category. It either does not stick or I get the error "Games that do not have IDs cannot belong to categories".

I am not sure what you intended here, but I think it's bugged yet.

**Comment by strycore** [comment]
What do you mean by "entirely uninstalled game"?  That wouldn't show up at all.

**Comment by danieljohnson2** [comment]
I mean a service (GOG) game that is in my GOG library, but which I have not installed.

**Comment by strycore** [comment]
Yes, that's a service game, it's a totally different beast that belongs in another table. That doesn't count as a game.

**Comment by danieljohnson2** [comment]
It's got a "Categories" command now.

**Comment by strycore** [comment]
That's because we're showing a contextual menu for non installed service games. This isn't correct. The contextual menu is meant for games only. On installed service games, it's ok to show it because there is an associated game entry.
For non installed service games, there should be a whole different type of menu or no menu at all (which I think was the behavior not long ago)

**Comment by danieljohnson2** [comment]
That was done intentionally, I am sure- the menu lets you install, and also 'View on Lutris.net'. I do not think we should remove it.

**Comment by strycore** [comment]
The view on Lutris entry isn't correct at all. It's slug based. I used it for Agony on GOG and it showed me the Agony for Amiga.

**Comment by danieljohnson2** [comment]
I *think* what the original issue wanted was to categories service games.

I notice that "Add installed game" also works rather badly on GOG games. That menu is rather half baked, but I think it is wanted if only it would work right.

**Comment by strycore** [comment]
No, that doesn't make sense in the slightest. Pretty sure the issue was about uninstalled games. 

We can't add meta-data to something that is an entirely different entity.

I really think we should have both a GameActions and ServiceGameActions 
That would also let us add an action like "View on GOG" or "View on the Epic Store".

**Comment by danieljohnson2** [comment]
"Add to favorites" is also available on service games; it just crashes Lutris for me. I do not like your commit, sir.

**Comment by strycore** [comment]
My commit is correct, it's showing the contextual menu on non installed service games that isn't.

**Comment by danieljohnson2** [comment]
Would you like me to take a crack at a second 'service games actions' menu?

**Comment by strycore** [comment]
Sure, I'm on a Steam Deck right now so I can't test things as easily as usual.

**Comment by danieljohnson2** [comment]
'kay. It won't have Categories in it; I still don't think that'll work. You can add that functionality later, if you see way to do it.

**Comment by strycore** [comment]
It's not meant to have categories. If anything, we would have to use the categories from the external service.

I wish there was a better way to signify that service games are not games.

**Comment by danieljohnson2** [comment]
We should rename them "Service Jobs". :laughing:

**Comment by danieljohnson2** [comment]
I've committed the new Service Games Actions code. It's not as big as it looks - a lot of small touches on the services, because I needed a way to ask 'what slug will you give this game when it's installed' so 'View on Lutris.net' can work. I also added 'what runner will you use for this game' so 'Locate Installed Game' can be smarter.

We can have a nice holy war about what Lutris service games should do for multi-platform games; my initial effort gives up if there's >1 platform, and the user must choose.

**Comment by BeatLink** [comment]
Whoah, lots of activity since I submitted. But yes, I was mainly hoping to use the categories for my service games such as Steam and Epic Games
